src/gulpfile.js

```diff
@@ -33,19 +61,50 @@ function generateComponentFiles (fullTemplatePath, componentPath, name, componen
     'ComponentTemplate/**/*.**'
   )
 
-  return gulp.src(resolvedTemplatePath)
-    .pipe(template({
-      name: name,
-      componentType: componentType ? toTitleCase(componentType) : 'Component',
-      upperCamelCaseName: upperCamelCaseName,
-      lowerCamelCaseName: toFirstLetterLowerCase(toUpperCamelCase(name)),
-      destinationPath: relativeDestinationPath,
-    }))
-    .pipe(rename((path) => {
-      path.basename = path.basename.replace('ComponentTemplate', upperCamelCaseName)
-    }))
-    .pipe(gulp.dest(destinationPath))
-    .on('end', function(){ console.log(`Component '${destinationPath}' was created.`) })
+  const endMessage = `Component '${destinationPath}' was created.`
+  return generateFiles(resolvedTemplatePath, name, componentType, upperCamelCaseName, relativeDestinationPath, destinationPath, endMessage)
+}
+
+/**
+ * @param {string} fullTemplatePath
+ * @param {string} componentPath
+ * @param {string} name
+ * @param {string} componentType
+ * @param {string} flavour
+ * @param {Array<string>} availableFlavours
+ */
+function generateFilesIfNotExistAlready (fullTemplatePath, componentPath, name, componentType, flavour, availableFlavours) {
+  const upperCamelCaseName = toUpperCamelCase(name)
+  const relativeDestinationPath = componentType ? path.join(componentType, upperCamelCaseName) : upperCamelCaseName
+  const destinationPath = path.join(componentPath, relativeDestinationPath)
+
+  validateDirectoryExists(destinationPath) // throws error if failed
+
+  const effectiveFlavour = (flavour || 'default').trim()
+
+  if (!availableFlavours.includes(effectiveFlavour)) {
+    throw new Error(`flavour '${effectiveFlavour}' does not exist, choose one of: ${availableFlavours}`)
+  }
+
+  const existingFiles = getFiles(destinationPath)
+  const existingFileExtensions = existingFiles.map(fileName => fileName.split(upperCamelCaseName + '.')[1])
+
+  const templateFiles = getFiles(path.join(fullTemplatePath, effectiveFlavour, 'ComponentTemplate'))
+  const templateFilesExtensions = templateFiles.map(fileName => fileName.split('ComponentTemplate.')[1])
+
+  templateFilesExtensions.forEach((extension) => {
```

Comment:
we should also support files without ComponentTemplate naming convention like README.md

Reply:
done